test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -0,0 +1,323 @@
+		vsc:      resource.vsc,
+		dInfo:    driver.GetDriverInfo(),
+	}
+
+	if driver.GetDriverInfo().Config.ClientNodeName != "" {
+		input.testCase.NodeName = driver.GetDriverInfo().Config.ClientNodeName
+	}
+
+	return resource, input
+}
+
+func (s *snapshottableTestSuite) execTest(driver TestDriver, pattern testpatterns.TestPattern) {
+	Context(getTestNameStr(s, pattern), func() {
+		var (
+			resource     snapshottableTestResource
+			input        snapshottableTestInput
+			needsCleanup bool
+		)
+
+		BeforeEach(func() {
+			needsCleanup = false
+			// Skip unsupported tests to avoid unnecessary resource initialization
+			skipUnsupportedTest(s, driver, pattern)
+			needsCleanup = true
+
+			// Create test input
+			resource, input = createSnapshottableTestInput(driver, pattern)
+		})
+
+		AfterEach(func() {
+			if needsCleanup {
+				resource.cleanupResource(driver, pattern)
+			}
+		})
+
+		// Ginkgo's "Global Shared Behaviors" require arguments for a shared function
+		// to be a single struct and to be passed as a pointer.
+		// Please see https://onsi.github.io/ginkgo/#global-shared-behaviors for details.
+		testSnapshot(&input)
+	})
+}
+
+type snapshottableTestResource struct {
+	driver    TestDriver
+	claimSize string
+
+	sc  *storage.StorageClass
+	pvc *v1.PersistentVolumeClaim
+	// volume snapshot class
+	vsc *unstructured.Unstructured
+}
+
+var _ TestResource = &snapshottableTestResource{}
+
+func (s *snapshottableTestResource) setupResource(driver TestDriver, pattern testpatterns.TestPattern) {
+	// Setup snapshottableTest resource
+	switch pattern.SnapshotType {
+	case testpatterns.DynamicCreatedSnapshot:
+		if dDriver, ok := driver.(DynamicPVTestDriver); ok {
+			s.sc = dDriver.GetDynamicProvisionStorageClass("")
+			if s.sc == nil {
+				framework.Skipf("Driver %q does not define Dynamic Provision StorageClass - skipping", driver.GetDriverInfo().Name)
+			}
+			s.driver = driver
+			s.claimSize = dDriver.GetClaimSize()
+			s.pvc = getClaim(s.claimSize, driver.GetDriverInfo().Config.Framework.Namespace.Name)
+			s.pvc.Spec.StorageClassName = &s.sc.Name
+			framework.Logf("In creating storage class object and pvc object for driver - sc: %v, pvc: %v", s.sc, s.pvc)
+
+			if sDriver, ok := driver.(SnapshottableTestDriver); ok {
+				s.vsc = sDriver.GetSnapshotClass()
+			}
+		}
+
+	default:
+		framework.Failf("Dynamic Snapshot test doesn't support: %s", pattern.SnapshotType)
+	}
+}
+
+func (s *snapshottableTestResource) cleanupResource(driver TestDriver, pattern testpatterns.TestPattern) {
+}
+
+type snapshottableTestInput struct {
+	testCase SnapshotClassTest
+	cs       clientset.Interface
+	dc       dynamic.Interface
+	pvc      *v1.PersistentVolumeClaim
+	sc       *storage.StorageClass
+	// volume snapshot class
+	vsc   *unstructured.Unstructured
+	dInfo *DriverInfo
+}
+
+func testSnapshot(input *snapshottableTestInput) {
+	It("should create snapshot with defaults", func() {
+		if input.dInfo.Name == "csi-hostpath-v0" {
```

Comment:
I think this check is now redundant. But I'm in favor of merging it as-is anyway and will remove it when moving the test setup around.

Reply:
@wackxu as @msau42 pointed out one more thing that needs to be changed before merging (the feature tag), can you remove this check while updating the PR?